src/ai/defaultai.cc

```diff
@@ -6297,21 +6271,15 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 	} else if (purpose == PerfEvaluation::kForDismantle) {  // now for dismantling
 		// never dismantle last building (a care should be taken elsewhere)
 		assert(bo.total_count() > 0);
-		if (bo.total_count() == 1) {
-			return BuildingNecessity::kNeeded;
-		} else if (bo.max_preciousness >= 10 && bo.total_count() == 2) {
-			return BuildingNecessity::kNeeded;
-		} else if (!bo.ware_outputs.empty() &&
-		           bo.current_stats > (10 + 60 / bo.ware_outputs.size()) / 2) {
-			return BuildingNecessity::kNeeded;
-		} else if (bo.inputs.size() == 1 &&
-		           calculate_stocklevel(static_cast<size_t>(bo.inputs.at(0))) >
-		              static_cast<unsigned int>(
-		                 std::abs(management_data.get_military_number_at(171)))) {
-			return BuildingNecessity::kNeeded;
-		} else {
-			return BuildingNecessity::kNotNeeded;
-		}
+
+		return (bo.total_count() == 1 || (bo.max_preciousness >= 10 && bo.total_count() == 2) ||
```

Comment:
Maybe still use `if` / `else`. Currently it is hard to find the actual values after this long condition. Or does clang complain about it?